src/functions.php

```diff
@@ -265,9 +265,15 @@ function task(bool $allowNull = false): ?Command
     return Container::get()->getCommand($allowNull);
 }
 
-function fs(): Filesystem
+function fs(?Context $context = null): ContextAwareFilesystem
 {
-    return Container::get()->fs;
+    $container = Container::get();
+    $context ??= $container->contextRegistry->getCurrentContext();
+
+    return new ContextAwareFilesystem(
+        $container->fs,
+        $context->workingDirectory,
+    );
 }
```

Comment:
IMO, this is a BC-break because it will change behaviour for people already using the fs() function and having defined a workingDirectory in their context.

We must either:
- introduce a new function and keep the existing fs() as is
- or have a mecanism to enable this context-dependant change, at project/context level (like an env var, or so)
- or maybe an additional `?bool` param, f.e. something that would be:
    - null by default, and trigger a deprecation notice to force user to choose a mode;
    - false to disable the context's wd
    - true to enable this new behaviour

Reply:
I'm not against a config object, but IMO it doesn't need a special attribute, just put it in the context : 

```
$context->config->hasContextAwareFilesystem
```

And to update it, use the default context / context attribute with something like

```
#[AsContext(default: true)]
function default_context(): Context
{
    return new Context(config: new Config(hasContextAwareFilesystem: true));
}
```

Also i don't think it should be a feature flag, because in this case we only have 2 choice : 

 * present
 * not present
 
Or we need 3 choices to handle BC break if we want this to be the default behavior in the future : 

 * true -> enable
 * false -> disable
 * null -> deprecated, you must choose a behavior, default to false, will default to true in next version